src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1577,57 +1577,56 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Sum")>]
         let inline sum (source: seq< (^a) >) : ^a = 
-            use e = source.GetEnumerator() 
             let mutable acc = LanguagePrimitives.GenericZero< (^a) >
-            while e.MoveNext() do
-                acc <- Checked.(+) acc e.Current
+            for item in source do
+                acc <- Checked.(+) acc item
```

Comment:
Please take a look at the generated IL (and do perf tests). The value type enumerator can't be utilized when the input type is only IEnumerable (a reference type) - it isn't even visible to the compiler. The enumerator will be boxed to IEnumerator  But take a look at the IL to check.